cpu.go

```diff
@@ -0,0 +1,64 @@
+package main
+
+import (
+	"fmt"
+	"runtime"
+	"sync"
+	"time"
+)
+
+func main() {
+	tensOfMillionsToCountTo := []int{9, 5, 11, 7}
+	runCpuHeavySync(tensOfMillionsToCountTo)
+	//sets number of available CPUs ==> 1
+	runtime.GOMAXPROCS(1)
+	runCpuHeavyGoroutines(tensOfMillionsToCountTo, 1)
+	//sets number of available CPUs ==> 8
+	runtime.GOMAXPROCS(8)
+	runCpuHeavyGoroutines(tensOfMillionsToCountTo, 8)
+}
+
+func runCpuHeavySync(tensOfMillionsToCountTo []int) {
+	t := time.Now()
+	for _, v := range tensOfMillionsToCountTo {
+		countToNMillion(v)
+	}
+	timeElapsed := time.Since(t)
+	fmt.Printf("Synchronous function completed all counts in %v\n", timeElapsed)
+}
+
+func runCpuHeavyGoroutines(tensOfMillionsToCountTo []int, cpus int) {
+	fmt.Printf("\nNumber of CPUs available to function: %v\n", cpus)
+	c := make(chan string, len(tensOfMillionsToCountTo))
```

Comment:
Seeing as this is a learning thing, be nice to have some notes / links about whats going on with channels and go routines here?